# Simulation settings for using moveit_sim_controllers
moveit_sim_hw_interface:
  joint_model_group: myrobot
  joint_model_group_pose: home
# Settings for ros_control_boilerplate control loop
generic_hw_control_loop:
  loop_hz: 300
  cycle_time_error_threshold: 0.01
# Settings for ros_control hardware interface
myrobot:
  hardware_interface:
    loop_hz: 10 # hz
    joints:
      - jt1_joint
      - jt2_joint
      - jt3_joint
      - jt4_joint
      - jt5_joint
    sim_control_mode: 1  # 0: position, 1: velocity
# Publish all joint states
# Creates the /joint_states topic necessary in ROS
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

#ros_control_namespace: /myrobot

controller_list:
  []
arm_controller:
  type: velocity_controllers/JointTrajectoryController
  joints:
    - jt1_joint
    - jt2_joint
    - jt3_joint
    - jt4_joint
  constraints:
    goal_time: 10.0
    stopped_velocity_tolerance: 0.1
    jt1_joint: {trajectory: 0.02, goal: 0.01}
    jt2_joint: {trajectory: 0.02, goal: 0.01}
    jt3_joint: {trajectory: 0.02, goal: 0.01}
    jt4_joint: {trajectory: 0.02, goal: 0.01}
  gains:
    #!!These values have not been optimized!!
    jt1_joint:  {p: 0.1,  i: 0.01, d: 0.0, i_clamp: 0.01}
    jt2_joint:  {p: 0.3,  i: 0.005, d: 0.05, i_clamp: 0.05}
    jt3_joint:  {p: 0.1,  i: 0.1, d: 0.1, i_clamp: 0.05}
    jt4_joint:  {p: 0.5,  i: 0.2, d: 0.1, i_clamp: 0.1}
  stop_trajectory_duration: 0.5
  state_publish_rate: 25
  action_monitor_rate: 10

gripper_controller:
  type: position_controllers/JointTrajectoryController
  joints:
    - jt5_joint




